modules/mobile/app/models/mobile/v2/adapters/appointments.rb

```diff
@@ -0,0 +1,328 @@
+            id: nil,
+            name: nil,
+            address: {
+              street: nil,
+              city: nil,
+              state: nil,
+              zip_code: nil
+            },
+            lat: nil,
+            long: nil,
+            phone: {
+              area_code: nil,
+              number: nil,
+              extension: nil
+            },
+            url: nil,
+            code: nil
+          }
+          telehealth = appointment_hash[:telehealth]
+
+          case type
+          when APPOINTMENT_TYPES[:cc]
+            cc_location = appointment_hash.dig(:extension, :cc_location)
+
+            if cc_location.present?
+              location[:name] = cc_location[:practice_name]
+              location[:address] = {
+                street: cc_location.dig(:address, :line)&.join(' ')&.strip,
+                city: cc_location.dig(:address, :city),
+                state: cc_location.dig(:address, :state),
+                zip_code: cc_location.dig(:address, :postal_code)
+              }
+            end
+          when APPOINTMENT_TYPES[:va_video_connect_atlas],
+            APPOINTMENT_TYPES[:va_video_connect_home],
+            APPOINTMENT_TYPES[:va_video_connect_gfe]
+            if telehealth
+              address = telehealth.dig(:atlas, :address)
+
+              if address
+                location[:address] = {
+                  street: address[:street_address],
+                  city: address[:city],
+                  state: address[:state],
+                  zip_code: address[:zip_code],
+                  country: address[:country]
+                }
+              end
+
+              location[:url] = telehealth[:url]
+              location[:code] = telehealth.dig(:atlas, :confirmation_code)
+            end
+          else
+            location[:id] = appointment_hash.dig(:location, :id)
+            location[:name] = appointment_hash.dig(:location, :name)
+            address = appointment_hash.dig(:location, :physical_address)
+            if address.present?
+              location[:address] = {
+                street: address[:line]&.join(' ')&.strip,
+                city: address[:city],
+                state: address[:state],
+                zip_code: address[:postal_code]
+              }
+            end
+            location[:lat] = appointment_hash.dig(:location, :lat)
+            location[:long] = appointment_hash.dig(:location, :long)
+            location[:phone] = location_phone(appointment_hash)
+          end
+
+          location
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        def location_phone(appointment_hash)
+          phone = appointment_hash.dig(:location, :phone, :main)
+          return nil unless phone
+
+          # captures area code (\d{3}) number (\d{3}-\d{4})
+          # and optional extension (until the end of the string) (?:\sx(\d*))?$
+          phone_captures = phone.match(/^(\d{3})-(\d{3}-\d{4})(?:\sx(\d*))?$/)
```

Comment:
didn't think of that. looking at the other phone regex methods, there are 3 total, but `modules/mobile/app/models/mobile/v0/adapters/community_care_appointments.rb` is slightly different but functionally the same. I think the only difference is it doesn't look for an explicitly for an `x` in the extension. Though since we're going to eventually remove the v0 adapters, eventually this will be the only phone method, so probably not worth doing the lib.